src/CVRF/cvrf_enumeration.c

```diff
@@ -0,0 +1,321 @@
+}
+
+const struct oscap_string_map CVRF_INVOLVEMENT_STATUS_MAP[] = {
+	{CVRF_INVOLVEMENT_OPEN, "Open"},
+	{CVRF_INVOLVEMENT_DISPUTED, "Disputed"},
+	{CVRF_INVOLVEMENT_IN_PROGRESS, "In Progress"},
+	{CVRF_INVOLVEMENT_COMPLETED, "Completed"},
+	{CVRF_INVOLVEMENT_CONTACT_ATTEMPTED, "Contact Attempted"},
+	{CVRF_INVOLVEMENT_NOT_CONTACTED, "Not Contacted"},
+	{0, NULL}
+};
+
+static cvrf_involvement_status_type_t cvrf_involvement_status_type_parse(xmlTextReaderPtr reader) {
+	return cvrf_enumeration_attr(reader, "Status", CVRF_INVOLVEMENT_STATUS_MAP);
+}
+
+const char *cvrf_involvement_status_type_get_text(cvrf_involvement_status_type_t involvement_type) {
+	return cvrf_enumeration_get_text(CVRF_INVOLVEMENT_STATUS_MAP, involvement_type);
+}
+
+const struct oscap_string_map CVRF_PRODUCT_STATUS_TYPE_MAP[] = {
+	{CVRF_PRODUCT_STATUS_FIRST_AFFECTED, "First Affected"},
+	{CVRF_PRODUCT_STATUS_KNOWN_AFFECTED, "Known Affected"},
+	{CVRF_PRODUCT_STATUS_KNOWN_NOT_AFFECTED, "Known Not Affected"},
+	{CVRF_PRODUCT_STATUS_FIRST_FIXED, "First Fixed"},
+	{CVRF_PRODUCT_STATUS_FIXED, "Fixed"},
+	{CVRF_PRODUCT_STATUS_RECOMMENDED, "Recommended"},
+	{CVRF_PRODUCT_STATUS_LAST_AFFECTED, "Last Affected"},
+	{0, NULL}
+};
+
+static cvrf_product_status_type_t cvrf_product_status_type_parse(xmlTextReaderPtr reader) {
+	return cvrf_enumeration_attr(reader, "Type", CVRF_PRODUCT_STATUS_TYPE_MAP);
+}
+
+const char *cvrf_product_status_type_get_text(cvrf_product_status_type_t product_status_type) {
+	return cvrf_enumeration_get_text(CVRF_PRODUCT_STATUS_TYPE_MAP, product_status_type);
+}
+
+
+const struct oscap_string_map CVRF_THREAT_TYPE_MAP[] = {
+	{CVRF_THREAT_IMPACT, "Impact"},
+	{CVRF_THREAT_EXPLOIT_STATUS, "Exploit Status"},
+	{CVRF_THREAT_TARGET_SET, "Target Set"},
+	{0, NULL}
+};
+
+static cvrf_threat_type_t cvrf_threat_type_parse(xmlTextReaderPtr reader) {
+	return cvrf_enumeration_attr(reader, "Type", CVRF_THREAT_TYPE_MAP);
+}
+const char *cvrf_threat_type_get_text(cvrf_threat_type_t threat_type) {
+	return cvrf_enumeration_get_text(CVRF_THREAT_TYPE_MAP, threat_type);
+}
+
+
+const struct oscap_string_map CVRF_REMEDIATION_TYPE_MAP[] = {
+	{CVRF_REMEDIATION_WORKAROUND, "Workaround"},
+	{CVRF_REMEDIATION_MITIGATION, "Mitigation"},
+	{CVRF_REMEDIATION_VENDOR_FIX, "Vendor Fix"},
+	{CVRF_REMEDIATION_NONE_AVAILABLE, "None Available"},
+	{CVRF_REMEDIATION_WILL_NOT_FIX, "Will Not Fix"},
+	{0, NULL}
+};
+
+static cvrf_remediation_type_t cvrf_remediation_type_parse(xmlTextReaderPtr reader) {
+	return cvrf_enumeration_attr(reader, "Type", CVRF_REMEDIATION_TYPE_MAP);
+}
+const char *cvrf_remediation_type_get_text(cvrf_remediation_type_t remediation_type) {
+	return cvrf_enumeration_get_text(CVRF_REMEDIATION_TYPE_MAP, remediation_type);
+}
+
+/*****************************************************************************
+ * CVRF Item Type
+ */
+
+struct cvrf_item_spec {
+	cvrf_item_type_t type;
+	const char *tag_name;
+	const char *container_name;
```

Comment:
This is used only for the static const struct cvrf_item_spec CVRF_ITEM_TYPE_MAP[] instances